common/classes/com/arjuna/common/internal/util/propertyservice/BeanPopulator.java

```diff
@@ -418,4 +418,8 @@ private static boolean isPositive(String value) {
     private static boolean isNegative(String value) {
         return "NO".equalsIgnoreCase(value) || "OFF".equalsIgnoreCase(value) || "FALSE".equalsIgnoreCase(value);
     }
+
+    public static void setBeanInstance(String name, Object beanInstance) {
+        beanInstances.putIfAbsent(name, beanInstance);
```

Comment:
Taking into account the comment from @ochaloup , I think if this is not going to force set the API and as it is an internal API I would expect an exception or boolean to report if the bean was set.

Reply:
Okay, I can return the previous bean value if it had already been set

Reply:
If it doesn't take effect some feedback needs to be given

Reply:
agree. I don't like the exceptions to be used for reporting something which is "expected" so using the boolean that returns if the operation succeeded is what I meant. The javadoc should be added for that too.

Reply:
Perhaps it is not expected that it would ever reach this state and returning the error would provide a meaningful condition for the calling code to react to.

e.g throw new Exception("This bean was already configured so your config will be ignored");

Reply:
but if we can internally reconcile the problem then a boolean would be more efficient

Reply:
another option is to reconfigure based on the new values in the beanInstance.

E.g. 
Object foo = beanInstances.putIfAbsent(name, beanInstance);
// overwrite values from foo with those from beanInstance

But in that case I think the method should return Object